pydra/engine/tests/test_shelltask_inputspec.py

```diff
@@ -537,6 +537,217 @@ def test_shell_cmd_inputs_mandatory_1():
     assert "mandatory" in str(e.value)
 
 
+def test_shell_cmd_inputs_template_1():
+    """ additional inputs, one uses output_file_template (and position)"""
+    my_input_spec = SpecInfo(
+        name="Input",
+        fields=[
+            (
+                "inpA",
+                attr.ib(
+                    type=str,
+                    metadata={
+                        "position": 1,
+                        "help_string": "inpA",
+                        "argstr": "",
+                        "mandatory": True,
+                    },
+                ),
+            ),
+            (
+                "outA",
+                attr.ib(
+                    type=str,
+                    metadata={
+                        "position": 2,
+                        "help_string": "outA",
+                        "argstr": "-o",
+                        "output_file_template": "{inpA}_out",
+                    },
+                ),
+            ),
+        ],
+        bases=(ShellSpec,),
+    )
+
+    shelly = ShellCommandTask(
+        executable="executable", input_spec=my_input_spec, inpA="inpA"
+    )
+    # outA has position, so is a part of the command line
+    assert shelly.cmdline == "executable inpA -o inpA_out"
+
+
+def test_shell_cmd_inputs_template_1a():
+    """ additional inputs, one uses output_file_template (without position)"""
+    my_input_spec = SpecInfo(
+        name="Input",
+        fields=[
+            (
+                "inpA",
+                attr.ib(
+                    type=str,
+                    metadata={
+                        "position": 1,
+                        "help_string": "inpA",
+                        "argstr": "",
+                        "mandatory": True,
+                    },
+                ),
+            ),
+            (
+                "outA",
+                attr.ib(
+                    type=str,
+                    metadata={
+                        "help_string": "outA",
+                        "argstr": "-o",
+                        "output_file_template": "{inpA}_out",
+                    },
+                ),
+            ),
+        ],
+        bases=(ShellSpec,),
+    )
+
+    shelly = ShellCommandTask(
+        executable="executable", input_spec=my_input_spec, inpA="inpA"
+    )
+    # outA has no position, so is not a part of the command line
```

Comment:
only `argst`r should indicate whether should indicate if something is added to the command line not `position`

Reply:
so you mean `argstr` could be not provided, is that right?

Reply:
yes - if you don't want it in the command line don't provide argstr (it will still be used for hashing, etc.,.).